router/broker.go

```diff
@@ -33,12 +35,38 @@ type subscription struct {
 	subscribers map[*wamp.Session]struct{}
 }
 
+// storedEvent is a wrapper around wamp event message with timestamp
+// to be used in event history
+type storedEvent struct {
+	timestamp    time.Time
+	Subscription wamp.ID
+	Publication  wamp.ID
+	Details      wamp.Dict
+	Arguments    wamp.List `wamp:"omitempty"`
+	ArgumentsKw  wamp.Dict `wamp:"omitempty"`
+}
+
+type eventHistoryItem struct {
+	publication *wamp.Publish
+	event       *storedEvent
```

Comment:
If you store this as a non-pointer your processor will like you better. Especially that all you store is another pointer and `eventHistoryItem` is also stored as a pointer array elsewhere.